metadata-ingestion/src/datahub/emitter/openapi_tracer.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import time
+from datetime import datetime, timedelta
+from typing import List
+
+from datahub.configuration.common import (
+    OperationalError,
+)
+from datahub.emitter.response_helper import TraceData
+
+logger = logging.getLogger(__name__)
+
+PENDING_STATUS = "PENDING"
+INITIAL_BACKOFF = 1.0  # Start with 1 second
+MAX_BACKOFF = 300.0  # Cap at 5 minutes
+BACKOFF_FACTOR = 2.0  # Double the wait time each attempt
+
+
+class OpenAPITrace:
+    def await_status(
+        self,
+        trace_data: List[TraceData],
+        trace_timeout: timedelta,
+    ) -> None:
+        """Verify the status of asynchronous write operations.
+
+        Args:
+            trace_data: List of trace data to verify
+            trace_timeout: Maximum time to wait for verification.
+
+        Raises:
+            OperationalError: If verification fails or times out
```

Comment:
we should create more specific exception classes instead of raising a generic OperationalError